api/aggregators/searchAggregator.js

```diff
@@ -93,6 +94,35 @@ exports.createMatchAggr = async (schemaName, projectId, keywords, caseSensitive,
   return aggregation;
 };
 
+exports.createKeywordRegexAggr = function(decodedKeywords, schemaName) {
+  let keywordRegexFilter = [];
+  // if we have a keyword search, and it is not wrapped in quotes (ie, phrase searching)
+  // then do a regex match. To help keep regex matches closer to their values, also
+  // filter on the score
+  if (decodedKeywords && !decodedKeywords.startsWith("\"") && !decodedKeywords.endsWith("\"")) {
+    // decodedKeywords is a const, so split, then join on the result or it'll
+    // throw an error about const assignment. Leave decodedKeyword immutable.
+    let terms = decodedKeywords.split(' ');
+    let searchTerm = terms.join('|');
+
+    // By default, if we're doing a keyword search exclude
+    // any values that have a score less then 5000.
+    let regexMatch = { $match: { score: { $gt: 5000 } } };
+
+    let regex = { $regex:'(?:^|(?<= ))(' + searchTerm + ')(?:(?= )|$)', $options:'i' };
+
+    if (schemaName === constants.PROJECT) {
```

Comment:
Should this live in this function, or should this function only create the Regex and not care about where it is being used?

Reply:
Thought about that. I guess I'm fairly indifference. My thought was other places might want specific regex functions for their tasks, so we'd wind up writing it again anyway